Foundation/NSStream.swift

```diff
@@ -202,43 +337,62 @@ public class NSOutputStream : Stream {
     }
     
     public override var streamStatus: Status {
-        return Stream.Status(rawValue: UInt(CFWriteStreamGetStatus(_stream)))!
+        return NSStream.Status(rawValue: UInt(CFWriteStreamGetStatus(_stream)))!
     }
     
     public class func outputStreamToMemory() -> Self {
         return self.init(toMemory: ())
     }
     
-    public override func propertyForKey(_ key: String) -> AnyObject? {
-        return CFWriteStreamCopyProperty(_stream, key._cfObject)
+    public override var streamError: NSError?{
+        let error = CFWriteStreamCopyError(_stream)
+        return error?._nsObject
     }
     
     public  override func setProperty(_ property: AnyObject?, forKey key: String) -> Bool {
         return CFWriteStreamSetProperty(_stream, key._cfObject, property)
     }
+    
+    public override func propertyForKey(_ key: String) -> AnyObject? {
+        return CFWriteStreamCopyProperty(_stream, key._cfObject)
+    }
+    
+    public func copy(with zone: NSZone?) -> AnyObject{
+        if let url = _url , let shouldAppend = _append {
+            return NSOutputStream(url: url, append: shouldAppend)!
+        } else if let buffer = _buffer, let capacity = _capacity {
+            return NSOutputStream(toBuffer: buffer, capacity: capacity)
+        } else {
+           return  NSOutputStream(toMemory: ())
+        }
+    }
+    
+    public func mutableCopy(with zone: NSZone?) -> AnyObject{
+        return copy(with: nil)
+    }
 }
 
 // Discussion of this API is ongoing for its usage of AutoreleasingUnsafeMutablePointer
 #if false
-extension Stream {
+extension NSStream {
     public class func getStreamsToHost(withName hostname: String, port: Int, inputStream: AutoreleasingUnsafeMutablePointer<InputStream?>?, outputStream: AutoreleasingUnsafeMutablePointer<NSOutputStream?>?) {
```

Comment:
I know the discussion of this api is ongoing, but I think it would be better to not prevent implementation due to a non-finalized api. I know many of us would love to be able to use this functionality on Linux. Here's a simple version I threw together that might satisfy the use case.

```
public class func getStreamsToHost(withName hostname: String, port: Int, inputStream: AutoreleasingUnsafeMutablePointer<InputStream?>?, outputStream: AutoreleasingUnsafeMutablePointer<NSOutputStream?>?) {
        var read: Unmanaged<CFReadStream>? = nil
        var write: Unmanaged<CFWriteStream>? = nil
        CFStreamCreatePairWithSocketToHost(kCFAllocatorDefault, host, UInt32(port), &read, &write)
        inputStream = read?.takeRetainedValue()
        outputStream = write?.takeRetainedValue()
    }
```